bridgebot/cardplay.py

```diff
@@ -91,19 +89,19 @@ def play(hands, contract, declarer):
                   __convert_hand_to_str(hands[current_player].hand)
                   )
 
-            legal_cards = hands[current_player].legal_cards(trump_suit)
+            legal_cards = hands[current_player].legal_cards(led_card.suit)
             print("Legal Cards: " +
                   __convert_hand_to_str(legal_cards)
                   )
 
             card = __get_card_from_user(legal_cards)
-            hands[current_player].follow(led_card, card)
+            hands[current_player].follow(led_card.suit, card)
             trick.append(card)
 
         # self.play_trick determines the index of the winner relative to the index of the leader
         # it also makes the code re-usable in case we want to use it for another trick-taking game, because
         # they all have the same mechanic, even if they don't have four players.
-        winning_player = (leading_player + determine_trick_winner(trick, trump_suit)) % 4
+        winning_player = leading_player.increment_by(determine_trick_winner(trick, strain))
```

Comment:
I had to define `increment_by` because it didn't like adding an `int` to a `Player`

Reply:
I like this idea, but I am unsure about the naming. The name implies that it is incrementing `leading_player` inplace instead of returning a value.